src/web/client/utility/UrlBuilder.ts

```diff
@@ -3,34 +3,61 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  */
 
-import { httpMethod, MULTI_ENTITY_URL_KEY, pathParamToSchema, SINGLE_ENTITY_URL_KEY } from "../common/constants";
+import { entityAttributesWithBase64Encoding, httpMethod, schemaEntityKey, schemaEntityName, schemaKey } from "../common/constants";
 import PowerPlatformExtensionContextManager from "../common/localStore";
+import { getEntity } from "./schemaHelper";
 
 export const getParameterizedRequestUrlTemplate = (isSingleEntity: boolean) => {
     const powerPlatformContext = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext();
     if (isSingleEntity) {
-        return powerPlatformContext.dataSourcePropertiesMap.get(SINGLE_ENTITY_URL_KEY) as string;
+        return powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.SINGLE_ENTITY_URL) as string;
     }
 
-    return powerPlatformContext.dataSourcePropertiesMap.get(MULTI_ENTITY_URL_KEY) as string;
+    return powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.MULTI_ENTITY_URL) as string;
 };
 
-export function getRequestURL(dataverseOrgUrl: string, entity: string, entityId: string, method: string, isSingleEntity: boolean): string {
+export function getRequestURL(
```

Comment:
Can we improve on the method name? This sounds like a GET request whereas it seems to generate a URL and return a string, also this file has a patchRequestURL function too

Reply:
We need to revisit naming convention in this extension as a whole. Out of scope for this PR.